view/src/test/java/cmm/apps/esmorga/view/navigation/NavigationTest.kt

```diff
@@ -224,10 +229,30 @@ class NavigationTest {
         composeTestRule.onNodeWithTag(EVENT_LIST_TITLE).assertIsDisplayed()
         composeTestRule.onNodeWithTag(EVENT_LIST_EVENT_NAME, true).performClick()
         composeTestRule.onNodeWithTag(EVENT_DETAILS_EVENT_NAME).assertIsDisplayed()
-        composeTestRule.onNodeWithTag(EventDetailsScreenTestTags.EVENT_DETAIL_PRIMARY_BUTTON, true).performClick()
+        composeTestRule.onNodeWithTag(EVENT_DETAIL_PRIMARY_BUTTON, true).performClick()
         composeTestRule.onNodeWithTag(LOGIN_TITLE).assertIsDisplayed()
     }
 
+    @Test
+    fun `given user in event details, when clicks on join event and api call fails, then error screen is shown`() {
+        val failureJoinEventUseCase = mockk<JoinEventUseCase>(relaxed = true).also { useCase ->
+            coEvery { useCase(any()) } returns EsmorgaResult.failure(Exception())
+        }
+        loadKoinModules(module { factory<JoinEventUseCase> { failureJoinEventUseCase } })
+
+        setNavigationFromDestination(Navigation.EventListScreen)
+
+        composeTestRule.onNodeWithTag(EVENT_LIST_TITLE).assertIsDisplayed()
```

Comment:
you do not need to add all this asserts in this test. They are checked in different test and this is duplicated.